### PR TITLE
Fix recent coverity issues

### PR DESCRIPTION
**Pull request by gaede**
@MarkusFrankATcernch  please check  logic in c6a818f.

### CODE REVIEW

File: DDG4/lcio/LCIOEventReader.cpp [hunk 1]
```diff
@@ -146,11 +146,10 @@ LCIOEventReader::readParticles(int event_number,
       vtx->y = p->vsy ;
       vtx->z = p->vsz ;
       vtx->time = p->time ;
-      haveVertex = false ;
+      haveVertex = true ;
```
Comment (petricm) [inline DDG4/lcio/LCIOEventReader.cpp]: This does not change anything. In line 86 `bool haveVertex = true ;` subsequently if the `if( !haveVertex && genStatus == 1 ){` is always false as describes in [CID 1402993](https://scan7.coverity.com/reports.htm#v17832/p11968/fileInstanceId=9708936&defectInstanceId=2393793&mergedDefectId=1402993). You have to change `if( !haveVertex && genStatus == 1 ){` to `if( haveVertex && genStatus == 1 ){` to address this.

Comment (petricm) [inline DDG4/lcio/LCIOEventReader.cpp]: Additionally  the `genStatus == 1` check does not cover all the cases, imagine haveing a K_L

**Comment by gaede** [comment]
Thanks for spotting this.
Concerning the genStat: you are right not all cases are covered here. But as discussed before, a complete and correct treatment needs more thought and work, eventually ( will create an issue about his)...

**Comment by petricm** [comment]
This addresses the control flow issue, but I don't think that `genStatus == 1` si a good choice, since you can also have non stable particles like K_L, for which the vertex will now be set to 0,0,0 but I don't think this should be the case.

**Comment by andresailer** [comment]
Frank , please consider this example
```
[00000001]   1|     -3122| 1.07e+01,-8.61e+00,-8.39e+00| 0.00e+00, 0.00e+00, 0.00e+00| 1.61e+01| 2 |[    0   ]| 0.00e+00, 0.00e+00, 0.00e+00| 2.08e+03,-1.67e+03,-1.63e+03| 1.12e+00| 0.00e+00| 0.00e+00, 0.00e+00, 0.00e+00|  (0, 0)   | [] - [2,3]
[00000002]   2|     -2212| 8.97e+00,-7.22e+00,-6.91e+00| 0.00e+00, 0.00e+00, 0.00e+00| 1.35e+01| 1 |[    0   ]| 2.08e+03,-1.67e+03,-1.63e+03| 0.00e+00, 0.00e+00, 0.00e+00| 9.38e-01|-1.00e+00| 0.00e+00, 0.00e+00, 0.00e+00|  (0, 0)   | [1] - []
[00000003]   3|       211| 1.74e+00,-1.39e+00,-1.48e+00| 0.00e+00, 0.00e+00, 0.00e+00| 2.68e+00| 1 |[    0   ]| 2.08e+03,-1.67e+03,-1.63e+03| 0.00e+00, 0.00e+00, 0.00e+00| 1.40e-01| 1.00e+00| 0.00e+00, 0.00e+00, 0.00e+00|  (0, 0)   | [1] - []
```
With genstat==1 the vertex would be set to `2.08e+03,-1.67e+03,-1.63e+03` and not to 0 0 0 as it should be. So either the vertex should be kept at 0 0 0 for the time being, or you need to implement a correct solution for all cases.

**Comment by gaede** [comment]
Ok, I agree the current solution is too simplistic. Unless you have a suggestion for a 'quick fix', I will revert to setting the vertex to (0,0,0) - as it was before. We have to be clear however, that this is really not correct . I have opened an issue to  keep track on this - comments/suggestions welcome.

**Comment by andresailer** [comment]
Thanks for reverting. I think this issue is much to complicated for a "quick fix", whoever takes this one should definitely write some test cases.

**Comment by andresailer** [comment]
Instead of merging from master, you need to rebase your branch on the current dd4hep/master branch, and then force push.

**Comment by gaede** [comment]
I did not intentionally merge but followed the link on GitHub to resolve the conflict that Markus had created with this changes  - not sure how to get out of this ...

**Comment by MarkusFrankATcernch** [comment]
I think I merged your branch.
Could you have a look if the changes are present?

**Comment by andresailer** [comment]
If Markus hadn't merged the branch you could have tried this:
Assuming the aidasoft/dd4hep remote is called `origin`, and your remote is called `downstream`
```
git stash ## store all local changes
git checkout master
git pull origin master
git checkout fix_coverity
git rebase master
## fix rebase conflicts, hopefully you have the git prompt installed
## use git status, git add , git rebase --continue, git rebase --skip, and the emacs M-x smerge-ediff 
git push -f downstream fix_coverity

```
Git prompt: https://github.com/andresailer/tutorial/blob/master/README.md#bashcompletion-and-gitprompt

Can we please stop merging and use rebase consistently.

**Comment by MarkusFrankATcernch** [comment]
I did:
git checkout -b gaede-fix_coverity master
git pull git://github.com/gaede/DD4hep.git fix_coverity

git checkout master
git merge --no-ff gaede-fix_coverity
git push origin master

**Comment by andresailer** [comment]
Please never do " merge --no-ff"

**Comment by MarkusFrankATcernch** [comment]
merge --no-ff is what the command line instructions say....

**Comment by andresailer** [comment]
I know, but the command line instructions are not for the "rebase and merge" way we are doing things.

**Comment by MarkusFrankATcernch** [comment]
So .... did the merge got through or not?
Can we close this or is there still a pull pending?

**Comment by andresailer** [comment]
We lost all of Frank's commit messages, but his changes are present.

**Comment by gaede** [comment]
I am in the middle of a rebase to resolve conflicts - so please do not close this - and also do not push anything ...

**Comment by gaede** [comment]
... and no - my changes are not in yet, as far as I can see.

**Comment by andresailer** [comment]
https://github.com/AIDASoft/DD4hep/commit/729e4baa3771459d5cc23da6fd999a11e91cf2dd

**Comment by gaede** [comment]
Ok, I see.  - How can we now fix the history ???

**Comment by MarkusFrankATcernch** [comment]
History we do not care -- we are looking forward ;-)
Is the code in? This is what matters....

**Comment by gaede** [comment]
Yes, code's in, history is screwed. Fine this time but I really would like to understand what went wrong and how git makes it even possible to drop history so easily.

**Comment by andresailer** [comment]
a) Basically there is a new commit that included all your changes. It would be easy to get the history back. You push your rebased branch and we kill the last two commits from the master branch...
b) Could we stop the self-fulfilling prophecy: "Nobody looks at commit messages because they are useless, so I don't write useful commit messages..."

**Comment by whit2333** [comment]
My 2 cents: I had similar issues using `git rebase` in the past. Unless you **really** want a nice linear history of fast forwards, it is best to avoid `rebase` unless you know exactly what you are doing.

**Comment by andresailer** [comment]
We *really* want a nice linear history. The only thing to avoid is directly pushing to the master branch unless you _really_ _really_ know what your are doing. ;-)

**Comment by gaede** [comment]
I agree, history is important and the goal should be to have a nice linear history - with meaningful commit messages. Ideally one should even be able to automatically create release notes from the commit messages ( ok, I know this is asking a lot ;-) ).
Anyways, how do I rebase to an individual commit ?

**Comment by gaede** [comment]
>You push your rebased branch and we kill the last two commits from the master branch...

Ok, I pushed a rebased version of `fix_coverity` - mainly as a git exercise for myself.
Seems to have worked nicely...
We could now kill the last _three_ (actually) commits and then do the usual rebase-merge. 
Not sure if this creates more problems with peoples working copies than is worth for fixing the history, though...

**Comment by andresailer** [comment]
You should have (force) pushed to your fork not create a new branch on the aidasoft one

**Comment by petricm** [comment]
All the changes are now in master

**Pull request closed without merging**